impeller/renderer/backend/vulkan/command_pool_vk.cc

```diff
@@ -92,8 +92,15 @@ bool CommandPoolVK::IsValid() const {
 
 void CommandPoolVK::Reset() {
   Lock lock(buffers_to_collect_mutex_);
-  GarbageCollectBuffersIfAble();
   graphics_pool_.reset();
+
+  // When the command pool is destroyed, all of its command buffers are freed.
+  // Handles allocated from that pool are now invalid and must be discarded.
+  for (vk::UniqueCommandBuffer& buffer : buffers_to_collect_) {
+    buffer.release();
```

Comment:
But the handle to the buffer will still be leaked right? Unless this is a reset.

Reply:
Earlier this function calls `graphics_pool_.reset()`, which destroys the Vulkan command pool and frees all of the pool's buffers.

After that the `buffer` handle is invalid, and the Vulkan validation layers will warn about a call to `vkFreeCommandBuffers` using the handle.